### ISSUE TITLE
Should the booleanAdapter have an empty selectors object?

### ISSUE THREAD

**Issue by andsilmar** [state-adapt/state-adapt#63]
Hi Mike! 

First I want to thank you for this great library! 

I want to create a super simple `showPasswordAdapter`  using your `booleanAdapter` like this: 

```typescript
showPasswordAdapter = buildAdapter<boolean>()(booleanAdapter)({
    vm: (state) => ({
      inputType: state ? 'text' : 'password',
      icon: state ? 'visibility' : 'visibility_off',
    }),
})();
```

But it won't work. Unless I'm missing something, I think the `booleanAdapter` should have an empty selector object in order to allow the composition that I'm looking for. 

```typescript
booleanAdapter = createAdapter<boolean>()({
    setTrue: () => true,
    setFalse: () => false,
    toggle: (state) => !state,
    selectors: {} // <- Adding this makes the example work
});
```

Thank you for you help!

**Comment by mfp22** [comment]
I'm glad you like it. Thanks.

I have seen this before and thought i fixed it. Maybe it was a similar but different scenario. I add tests every time I fix stuff, so I probably didn't fix this specific issue. 

2 things from your code though
1. It's not really `state` that's being passed in. It's the derived states returned from the adapter's selectors. You could call it that if you want:

```typescript 
const showPasswordAdapter = buildAdapter<boolean>()(booleanAdapter)({
    vm: selectorStates => ({
      inputType: selectorStates.state ? 'text' : 'password',
      icon: selectorStates.state ? 'visibility' : 'visibility_off',
    }),
})();
```

But I just prefer `s`:

```typescript
const showPasswordAdapter = buildAdapter<boolean>()(booleanAdapter)({
    vm: s => ({
      inputType: s.state ? 'text' : 'password',
      icon: s.state ? 'visibility' : 'visibility_off',
    }),
})();
```

The 2nd thing is, I've never understood the point of `vm`. I remember doing this in AngularJS because it was a sort of standard practice for `controllerAs` syntax or something. But in Angular it groups things together that I don't think need to be grouped. In this case, if you add more properties to that object being returned, it will recalculate everything anytime any dependency updates. It's not an efficient pattern. I would just define the selectors. 

```typescript
const showPasswordAdapter = buildAdapter<boolean>()(booleanAdapter)({
  inputType: s.state ? 'text' : 'password',
  icon: s.state ? 'visibility' : 'visibility_off',
})();
```

If you like, maybe you can call the whole store that gets created `vm`. 

---

I'm going to look at the TypeScript issue now though. Thanks for opening this issue.

**Comment by andsilmar** [comment]
Thank you for your response Mike! 

Yeah, I totally agree with your comments regarding the `vm`. I just use it when all the selectors depend on a single property like the example, because when that property changes, all the selectors should change like my example. In that way I can "save" a subscription in the template.

Also, I tend to do this when I'm working with third party components with inputs that complain about the value being undefined.

Do you see a performance issue in that case also? This is my particular use case:

```html
<!-- service.showPassword = adapt(false, passwordAdapter) -->
<mat-form-field *ngIf="service.showPassword.vm$ | async as vm">
    <mat-label>Contraseña</mat-label>
    <input
      [type]="vm.inputType"
      matInput
      formControlName="password"
      required
      data-testid="password"
    />
    <button
      type="button"
      mat-icon-button
      matSuffix
      [attr.aria-label]="'Ocultar contraseña'"
      data-testid="password-toggler"
      (click)="service.showPassword.toggle()"
    >
      <mat-icon> {{ vm.icon }} </mat-icon>
    </button>
    <mat-error appFirstError> </mat-error>
  </mat-form-field>
```

Thank you!

**Comment by mfp22** [comment]
I don't think it's worth it for the performance. `vm` is an invented concept that I think is just overhead. What I would do about the `async` annoyance is either use RxAngular, wait for StateAdapt 2.1 which will have signals, or define the `vm` inline:

```html
*ngIf="{ prop: store.prop$ | async, prop2: store.prop2$ | async } as vm"
```

This would be more performant and would be easy to replace once signals are in. 

For signals, the need for an initial value would cause a problem if I didn't already have to solve this in React. For now, the signal will use the `initialState` from the store, unless it has already been initialized. So the initialization issue should not exist for StateAdapt 2.1.

**Comment by mfp22** [comment]
Oh, and I have a fix for this main issue with `booleanAdapter` and adapters in general when used in `buildAdapter` within a week. For now, the workaround should be clear (although unsatisfying).

**Comment by andsilmar** [comment]
Thank you Mike for your comments and work! I'm eager to see StateAdapt 2.1 🙌

**Comment by mfp22** [comment]
Fixed with https://github.com/state-adapt/state-adapt/commit/c26328f46a2b1e3ba26a37f3c18b348890c794af released in 2.0.5
